### PR TITLE
feat(core): add NX_ENV_EXPAND environment variable to disable expansion if desired

### PR DESCRIPTION
**Pull request by benpsnyder**
## Current Behavior
Nx always expands environment variables using `dotenv-expand`.

## Expected Behavior
The user can disable expansion of variables by setting `NX_ENV_EXPAND=false`

## Related Issue(s)
#19400

### CODE REVIEW

**Comment by Cammisuli** [comment]
Sorry! I've been busy working on something that's taking more time than I would have liked 😅 

I haven't forgot about this, it's on my todos!

File: packages/nx/src/tasks-runner/task-env.ts [hunk 1]
```diff
@@ -124,7 +124,10 @@ function getNxEnvVariablesForTask(
 
 function loadDotEnvFilesForTask(
   task: Task,
-  environmentVariables: NodeJS.ProcessEnv
+  environmentVariables: NodeJS.ProcessEnv,
+  envExpand = process.env.NX_ENV_EXPAND !== undefined
```
Comment (Cammisuli) [inline packages/nx/src/tasks-runner/task-env.ts:128]: this doesn't need to be a argument to this function. You can remove it and use the logic suggested in the if branch

File: packages/nx/src/tasks-runner/task-env.ts [hunk 2]
```diff
@@ -172,13 +175,20 @@ function loadDotEnvFilesForTask(
...
       override: false,
     });
-    environmentVariables = {
-      ...expand({
-        ...myEnv,
-        ignoreProcessEnv: true, // Do not override existing env variables as we load
-      }).parsed,
-      ...environmentVariables,
-    };
+    if (envExpand) {
```
Comment (Cammisuli) [inline packages/nx/src/tasks-runner/task-env.ts:178]:
```suggestion
    if (process.env.NX_ENV_EXPAND !== 'false') {
```

**Review by Cammisuli — CHANGES_REQUESTED** [review]
see comments

**Comment by SmoshySmosh** [comment]
I'm really looking forward to this feature!

**Comment by Riron** [comment]
If I am not mistaking, the same check should be applied [here](https://github.com/nrwl/nx/blob/master/packages/nx/bin/nx.ts#L119). Otherwise, NX will be comparing expanded & non expanded envs when unloading envs during tasks run [here](https://github.com/nrwl/nx/blob/master/packages/nx/src/tasks-runner/task-env.ts#L192).

**Pull request closed without merging**